### DISCUSSION TITLE
goto doesn't work

### DISCUSSION THREAD

**Discussion by ArcherSean** [yuin/gopher-lua#510] [Q&A]
You **must** post issues only here. Questions, ideas **must** be posted in [discussions](https://github.com/yuin/gopher-lua/discussions). 

- [ ] GopherLua is a Lua5.1 implementation. You should be familiar with Lua programming language. Have you read [Lua 5.1 reference manual](http://www.lua.org/manual/5.1/) carefully?
- [ ] GopherLua is a Lua5.1 implementation.  In Lua, to keep it simple, **it is more important to remove functionalities rather than to add functionalities unlike other languages** . If you are going to introduce some new cool functionalities into the GopherLua code base and the functionalities can be implemented by existing APIs, It should be implemented as a library. 

Please answer the following before submitting your issue:

1. What version of GopherLua are you using? : v1.1.1
2. What version of Go are you using? : 1.21
3. What operating system and processor architecture are you using? :MacOS/ARM
4. What did you do? :use goto and ::label::
5. What did you expect to see? : 'goto' will jump to ::label::
6. What did you see instead? :syntax error

**Comment by ArcherSean** [comment]
Hi there, I met a problem when I use 'goto' in gopher-lua v1.1.1. When I use goto like this
   ```
for i = 1, #backupSource do
     if i == 2 then
        goto continue
    end
    ::continue::
end
```

It get error:
`near '::':   syntax error`
Is there something wrong with how I used it?

**Comment by xkisu** [comment]
I believe it's not supported, as this library is a Lua 5.1 implementation and `goto` [was added in 5.2 ](http://lua-users.org/wiki/GotoStatement)

**Comment by ArcherSean** [comment]
> I believe it's not supported, as this library is a Lua 5.1 implementation and `goto` [was added in 5.2 ](http://lua-users.org/wiki/GotoStatement)

But in the version v1.1.0, gopher-lua has supported `goto` 
![image](https://github.com/user-attachments/assets/f5eb4761-1aaf-41fd-9ba5-08e7fdd11bee)
